chart/openfaas/templates/oauth2-plugin-dep.yaml

```diff
@@ -0,0 +1,121 @@
+{{- $functionNs := default .Release.Namespace .Values.functionNamespace }}
+{{- if .Values.oauth2Plugin }}
+{{- if .Values.oauth2Plugin.enabled }}
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: {{ template "openfaas.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    component: oauth2-plugin
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  name: oauth2-plugin
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  replicas: {{ .Values.basicAuthPlugin.replicas }}
```

Comment:
I'm not a fan of using the basic auth plugin value here for the oauth plugin, I wonder if we could use a template helper to help move to a more generic name and also make this more semantic while maintaining backwards compatibility.

Reply:
Me neither, it's just a typo, not intentional.